### PR TITLE
Fix memory leaks,

### PR DESCRIPTION
**Pull request by patriotyk**
Instead of openssl use libsodium for hash functions

### CODE REVIEW

File: src/aktualizr.cc [hunk 1]
```diff
@@ -40,18 +40,23 @@ Aktualizr::Aktualizr(const Config &config) : config_(config) {
 #endif
 }
 
+Aktualizr::~Aktualizr() {
+  CRYPTO_cleanup_all_ex_data();
```
Comment (cajun-rat) [inline src/aktualizr.cc]: I'm struggling to find good documentation on this function, but what happens if someone creates two Aktualizr objects in the same application?

Would curl_global_cleanup (to match curl_global_init) fix the problem?

File: src/aktualizr.cc [hunk 2]
```diff
@@ -40,18 +40,23 @@ Aktualizr::Aktualizr(const Config &config) : config_(config) {
...
+}
+
 int Aktualizr::run() {
-  command::Channel *commands_channel = new command::Channel();
-  event::Channel *events_channel = new event::Channel();
+  command::Channel commands_channel;
+  event::Channel events_channel;
+
+  EventsInterpreter events_interpreter(config_, &events_channel, &commands_channel);
```
Comment (cajun-rat) [inline src/aktualizr.cc]: +1 to nuking 'new'

**Comment by patriotyk** [comment]
I have updated, now it is ready for review.

**Comment by cajun-rat** [comment]
I get the following error from the production server with this change:

{code}
[2017-07-31 11:43:49.718849]: response: {"code":"signature_not_valid","description":"The given signature is not valid"}
{code}

Does it work for you?

**Comment by cajun-rat** [comment]
I'm seeing the following test failures after rebasing against master:

~~~
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] SotaUptaneClientTest.device_registered
[  FAILED  ] SotaUptaneClientTest.RunForeverNoUpdates
[  FAILED  ] SotaUptaneClientTest.RunForeverHasUpdates
[  FAILED  ] SotaUptaneClientTest.RunForeverInstall

 4 FAILED TESTS
~~~

File: src/aktualizr.cc [hunk 3]
```diff
@@ -39,18 +39,21 @@ Aktualizr::Aktualizr(const Config &config) : config_(config) {
 #endif
 }
 
+Aktualizr::~Aktualizr() { CRYPTO_cleanup_all_ex_data(); }
```
Comment (cajun-rat) [inline src/aktualizr.cc]: Still not happy about this, but we can fix this another time

**Pull request merged**